### DISCUSSION TITLE
How to properly set an upstream proxy - Invalid proxy mode specification

### DISCUSSION THREAD

**Discussion by bizmate** [mitmproxy/mitmproxy#7154] [General]
I have posted a question here https://github.com/mitmproxy/mitmproxy/issues/3142#issuecomment-2325280294 but maybe it was not the right place. Copied content is below

-----------------
Hi, I can see this issue is closed, i can open a new one if necessary. I tried starsting the proxy with this docker compose setting

  proxy:
    image: mitmproxy/mitmproxy:10.4.2
    command: "mitmdump --mode upstream:${UPSTREAM_PROXY}"
the UPSTREAM_PROXY gets a value of http://{user}:{password}@host:port (also tried without the schema http) but i keep getting

  /usr/local/bin/mitmdump: Invalid proxy mode specification: upstream:BLABLAUSER:PASS@gate.superproxy.com:7000 (Invalid server specification: BLABLAUSER:PASS)

I cannot see these examples in the docs in order to correctly specify the authentication info. Can someone point to the right way of specifying the configuration?

**Comment by Prinzhorn** [comment]
`--upstream-auth`
